File: finder4j-backtrack-examples/src/test/java/com/github/K0zka/finder4j/backtrack/examples/lab/LabStateTest.java
package com.github.K0zka.finder4j.backtrack.examples.lab;

import java.util.Arrays;

import org.junit.Assert;
import org.junit.Test;

import com.github.K0zka.finder4j.backtrack.examples.lab.LabObject;
import com.github.K0zka.finder4j.backtrack.examples.lab.LabState;
import com.github.K0zka.finder4j.backtrack.examples.lab.Position;

public class LabStateTest {
	@Test
	public void isComplete() {
		LabState state = new LabState(new LabObject[][] { { LabObject.Exit } },
				Arrays.asList(new Position(0, 0)));
		Assert.assertTrue(state.isComplete());
	}

	@Test
	public void isCompleteNotComplete() {
		LabState state = new LabState(new LabObject[][] {
				{ LabObject.Floor, LabObject.Exit },
				{ LabObject.Floor, LabObject.Floor } },
				Arrays.asList(new Position(0, 0)));
		Assert.assertFalse(state.isComplete());
	}

	@Test
	public void testToString() {
		System.out.println(new LabState(25, 25));
	}

}

File: finder4j-backtrack-examples/src/test/kotlin/com/github/k0zka/finder4j/backtrack/examples/lab/LabStateTest.kt
package com.github.k0zka.finder4j.backtrack.examples.lab

import org.junit.Assert
import org.junit.Test
import java.util.Arrays
import java.util.Arrays.asList

class LabStateTest {
	@Test
	fun isComplete() {
		val state = LabState(arrayOf(arrayOf(LabObject.Exit)),
							 asList(Position(0, 0)))
		Assert.assertTrue(state.isComplete)
	}

	@Test
	fun isCompleteNotComplete() {
		val state = LabState(
				arrayOf(arrayOf(LabObject.Floor, LabObject.Exit), arrayOf(LabObject.Floor, LabObject.Floor)),
				Arrays.asList(Position(0, 0)))
		Assert.assertFalse(state.isComplete)
	}

	@Test
	fun testToString() {
		println(LabState(25, 25))
	}

}
